File: simple-stack-example-mvvm-fragments/src/main/java/com/zhuinden/simplestackexamplemvvm/application/injection/AndroidModule.java
package com.zhuinden.simplestackexamplemvvm.application.injection;

import android.app.Application;
import android.content.Context;
import android.content.res.Resources;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Zhuinden on 2017.07.25..
 */

@Module
public class AndroidModule {
    private final Application app;

    public AndroidModule(Application application) {
        this.app = application;
    }

    @Provides
    Application application() {
        return app;
    }

    @Provides
    Context context() {
        return app;
    }

    @Provides
    Resources resources() {
        return app.getResources();
    }
}

File: samples/legacy-architecture-samples/simple-stack-example-mvvm-fragments/src/main/java/com/zhuinden/simplestackexamplemvvm/application/injection/AndroidModule.kt
package com.zhuinden.simplestackexamplemvvm.application.injection

import android.app.Application
import android.content.Context
import android.content.res.Resources
import dagger.Module
import dagger.Provides

/**
 * Created by Zhuinden on 2017.07.25..
 */
@Module
class AndroidModule {
    @Provides
    fun context(application: Application): Context = application

    @Provides
    fun resources(application: Application): Resources = application.resources
}